apps/backend/src/database/database.module.ts

```diff
@@ -13,6 +35,41 @@ function getSynchronize(configService: ConfigService): boolean {
   }
 }
 
+function toObject(fields: string[], values?: string[]): Record<number, string> {
+  if (values) {
+    const result: Record<number, string> = {};
+    for (let i = 0; i < values.length; i++) {
+      if (!sensitiveKeys.some((regex) => regex.test(fields[i]))) {
+        result[i] = values[i];
+      } else {
+        result[i] = '[REDACTED]';
+      }
+    }
+    return result;
+  } else {
+    return {};
+  }
+}
+
+function logQuery(
+  sql: string,
+  connection: {fields: string[]; bind: string[]; type: string}
+) {
+  if (connection.type === 'INSERT' || connection.type === 'UPDATE') {
+    const matches = sql.match(/\$\d/gm);
+    const values = toObject(connection.fields, connection.bind);
+
+    if (matches) {
+      matches.forEach((placeholder, index) => {
+        sql = sql.replace(placeholder, `"${values[index]}"`);
+      });
+    }
```

Comment:
The placeholders should be left as-is so we can see if the SQL query was parameterized or not. This replacement taints the generated SQL.

If possible append the values to the end of the string.